src/node_buffer.cc

```diff
@@ -565,6 +566,53 @@ void StringSlice(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(ret);
 }
 
+// Convert the input into an encoded string
+void DecodeUTF8(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);  // list, flags
+
+  if (!(args[0]->IsArrayBuffer() || args[0]->IsSharedArrayBuffer() ||
+        args[0]->IsArrayBufferView())) {
+    return node::THROW_ERR_INVALID_ARG_TYPE(
+        env->isolate(),
+        "The \"list\" argument must be an instance of SharedArrayBuffer, "
+        "ArrayBuffer or ArrayBufferView.");
+  }
+
+  ArrayBufferViewContents<char> buffer(args[0]);
+
+  CHECK(args[1]->IsBoolean());
+  bool ignore_bom = args[1]->BooleanValue(env->isolate());
+
+  const char* data = buffer.data();
+  size_t beginning = 0;
+  size_t length = buffer.length();
+
+  if (buffer.length() == 0) return args.GetReturnValue().SetEmptyString();
```

Comment:
```suggestion
  if (length == 0) return args.GetReturnValue().SetEmptyString();
```

Reply:
Thank you. I'll update the pull request when the initial tests are complete.